### PR TITLE
Also test Redshift without S3 scratch bucket

### PR DESCRIPTION
**Pull request by vinceatbluelabs**
Now that we have full capability to use Redshift without an S3 bucket - let's add integration tests to ensure it keeps working!

This is based on #117 and set to merge into that branch for clarity of the diff.

### CODE REVIEW

File: tests/integration/records/single_db/test_records_load.py [hunk 1]
```diff
@@ -10,12 +10,15 @@
 
 class RecordsLoadIntegrationTest(BaseRecordsIntegrationTest):
     def load_and_verify(self, format_type, variant, hints={}, broken=False, sourcefn=None):
-        if self.engine.name == 'bigquery' and variant == 'csv':
+        redshift_with_no_bucket = self.engine.name == 'redshift' and not self.has_scratch_bucket()
+        # TODO: See if the issue I'm seeing in tests is related too
```
Comment (bluelabsbutler) [inline tests/integration/records/single_db/test_records_load.py:14]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: .circleci/config.yml [hunk 2]
```diff
@@ -585,14 +609,16 @@ workflows:
...
+            - redshift-itest-no-s3
+            - redshift-itest-s3
+            - redshift-itest-s3-old-pandas
+            - redshift-itest-s3-no-pandas
             - postgres-itest
             - mysql-itest
             - cli-1-itest
             - cli-2-itest
             - cli-3-itest
+            # TODO: Decide on name ordering
```
Comment (bluelabsbutler) [inline .circleci/config.yml:621]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: .circleci/config.yml [hunk 3]
```diff
@@ -274,6 +278,12 @@ jobs:
             export RECORDS_MOVER_SESSION_TYPE=env
             mkdir -p test-reports/itest
             cd tests/integration/records/single_db
+            # TODO: Apply same pattern in with_dbs version
```
Comment (bluelabsbutler) [inline .circleci/config.yml:281]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: .circleci/config.yml [hunk 4]
```diff
@@ -370,7 +379,7 @@ workflows:
           python_version: "3.6"
           pandas_version: "<1"
           requires:
-            - redshift-itest
+            - redshift-s3-itest
```
Comment (vinceatbluelabs) [inline .circleci/config.yml:382]: The existing tests in CircleCI use an S3 scratch bucket, so let's label them as such.

File: .circleci/config.yml [hunk 5]
```diff
@@ -524,24 +533,35 @@ workflows:
...
+          name: redshift-no-s3-itest
+          extras: '[redshift-binary,itest]'
+          python_version: "3.6"
+          db_name: demo-itest
+          include_s3_scratch_bucket: false
+          requires:
+            - redshift-s3-itest
+          filters:
+            tags:
+              only: /v\d+\.\d+\.\d+(-[\w]+)?/
```
Comment (vinceatbluelabs) [inline .circleci/config.yml:545]: The new test!  One little parameter swap.

File: metrics/bigfiles_high_water_mark [hunk 6]
```diff
@@ -1 +1 @@
-1049
+1061
```
Comment (vinceatbluelabs) [inline metrics/bigfiles_high_water_mark:1]: `tests/integration/itest` is large.  At some point it ought to be split up into modules, but I'm not entirely sure it's a net complexity win to do that yet.

File: tests/integration/itest [hunk 7]
```diff
@@ -255,6 +255,16 @@ def run_test(args, target, parser):
...
+    elif (target == 'redshift-no-s3'):
+        if (args.docker):
+            docker_compose_run(['with-db', 'demo-itest',
+                                'nosetests', '--xunit-file=nosetests.xml', '.'],
+                               prefixes=["with-aws-creds", "circleci"],
+                               cwd="/usr/src/app/tests/integration/records/single_db")
+        else:
+            subprocess.check_call(['with-db', 'demo-itest',
+                                   'nosetests', '--xunit-file=nosetests.xml', '.'],
+                                  cwd=f"{script_dir}/records/single_db")
```
Comment (vinceatbluelabs) [inline tests/integration/itest:267]: These are called into *without* calling `        with set_circleci_scratch_bucket():` (there's no bucket by default).

File: tests/integration/records/single_db/test_records_load.py [hunk 8]
```diff
@@ -10,12 +10,12 @@
...
-            # https://github.com/bluelabsio/records-mover/issues/80
+        redshift_with_no_bucket = self.engine.name == 'redshift' and not self.has_scratch_bucket()
+        if redshift_with_no_bucket:
             # https://github.com/bluelabsio/records-mover/issues/81
-            logger.warning("This test won't pass until we can use hints to "
-                           "infer date/time/etc columns, or we use records schema "
-                           "from target to set types on the source, so skipping.")
+            logger.warning("This test won't pass until we can use the "
+                           "records schema from the target to cast the "
+                           "dataframe types appropriately, so skipping.")
```
Comment (vinceatbluelabs) [inline tests/integration/records/single_db/test_records_load.py:18]: This is unfortunate, as it does keep us from doing a bunch of the load tests.  For what it's worth, while poring through the error messages when this ran without this guard, they all seemed to work just fine modulo the lack of Redshift TIME type.

File: tests/integration/records/table_timezone_validator.py [hunk 9]
```diff
@@ -115,12 +115,19 @@ def validate(self,
               self.tc.variant_doesnt_support_seconds(self.file_variant)))):
             seconds = '00'
             micros = '000000'
+
+            # Some databases on rendering this as a string will just
+            # drop the microseconds if they're all zeros.
```
Comment (vinceatbluelabs) [inline tests/integration/records/table_timezone_validator.py:120]: As noted in the logic above this, some CSV variants don't support milli/micro-seconds, and it looks like the increased testing here triggered exactly such a path with Redshift (yay, more testing!).

**Review by crvena-sonja — APPROVED** [review]
:shipit:

**Pull request merged**